source/beamline/REIXS/REIXSSampleMotor.cpp

```diff
@@ -0,0 +1,377 @@
+#include "REIXSSampleMotor.h"
+
+#include "beamline/CLS/CLSMDriveMotorControl.h"
+#include "actions3/AMListAction3.h"
+#include "actions3/AMActionSupport.h"
+#include "util/AMErrorMonitor.h"
+
+REIXSSampleMotor::REIXSSampleMotor(AMMotorGroupObject::MotionDirection direction,
+                                   const QString& name,
+                                   const QString& units,
+		                           AMControl* horizontalTranslationControl,
+		                           AMControl* normalTranslationControl,
+		                           AMControl* verticalRotationControl,
+		                           QObject *parent,
+		                           const QString& description) :
+
+    AMPseudoMotorControl(name, units, parent, description)
+{
+	direction_ = direction;
+
+	if(direction_ == AMMotorGroupObject::VerticalMotion) {
+
+		AMErrorMon::alert(this,
+		                  REIXS_SAMPLE_MOTOR_INVALID_DIRECTION,
+		                  QString("REIXS Sample Motor initialized in vertical axis"));
+	}
+
+	horizontalTranslationControl_ = horizontalTranslationControl;
+	normalTranslationControl_ = normalTranslationControl;
+	verticalRotationControl_  = verticalRotationControl;
+
+	angleOffset_ = 0;
+	connectedOnce_ = false;
+
+	addChildControl(horizontalTranslationControl_);
+	addChildControl(normalTranslationControl_);
+	addChildControl(verticalRotationControl_);
+
+	connect(verticalRotationControl_, SIGNAL(valueChanged(double)),
+	        this, SLOT(onVerticalAxisRotated(double)));
```

Comment:
This is connected up here because the recalculation of the coordinate system is only required when the verticalRotationControl_ alters its value. If I put the same code it would also happen when the either the horizontal and normal motors performed a translation (which doesn't require a recalculation of the system rotation).